include/cute/arch/copy_xe_spirv.hpp

```diff
@@ -0,0 +1,504 @@
+  __attribute__((opencl_global)) void *base_address, int width, int height, int pitch,
+  cute::intel::coord_t coord);
+
+SYCL_EXTERNAL
+void intel_sub_group_2d_block_prefetch_16b_2r16x2c(
+  __attribute__((opencl_global)) void *base_address, int width, int height, int pitch,
+  cute::intel::coord_t coord);
+
+SYCL_EXTERNAL
+void intel_sub_group_2d_block_prefetch_16b_4r16x2c(
+  __attribute__((opencl_global)) void *base_address, int width, int height, int pitch,
+  cute::intel::coord_t coord);
+
+SYCL_EXTERNAL
+void intel_sub_group_2d_block_prefetch_32b_16r8x1c(
+  __attribute__((opencl_global)) void *base_address, int width, int height, int pitch,
+  cute::intel::coord_t coord);
+
+// SPIRV copy definitions
+SYCL_EXTERNAL __attribute__((convergent)) void __spirv_Subgroup2DBlockLoadINTEL(
+  int ElementSize, int BlockWidth, int BlockHeight, int BlockCount,
+  const void *src_base_pointer, int memory_width, int memory_height,
+  int memory_pitch, cute::intel::coord_t coordinate, void *dst_pointer);
+
+SYCL_EXTERNAL __attribute__((convergent)) void __spirv_Subgroup2DBlockLoadTransformINTEL(
+  int ElementSize, int BlockWidth, int BlockHeight, int BlockCount,
+  const void *src_base_pointer, int memory_width, int memory_height,
+  int memory_pitch, cute::intel::coord_t coordinate, void *dst_pointer);
+
+SYCL_EXTERNAL __attribute__((convergent)) void __spirv_Subgroup2DBlockLoadTransposeINTEL(
+  int ElementSize, int BlockWidth, int BlockHeight, int BlockCount,
+  const void *src_base_pointer, int memory_width, int memory_height,
+  int memory_pitch, cute::intel::coord_t coordinate, void *dst_pointer);
+
+SYCL_EXTERNAL __attribute__((convergent)) void __spirv_Subgroup2DBlockStoreINTEL(
+  int ElementSize, int BlockWidth, int BlockHeight, int BlockCount,
+  void *src_pointer, const void *dst_base_pointer, int memory_width,
+  int memory_height, int memory_pitch, cute::intel::coord_t coordinate);
+
+SYCL_EXTERNAL __attribute__((convergent)) void __spirv_Subgroup2DBlockPrefetchINTEL(
+  int ElementSize, int BlockWidth, int BlockHeight, int BlockCount,
+  const void *src_base_pointer, int memory_width, int memory_height,
+  int memory_pitch, cute::intel::coord_t coordinate);
+
+namespace cute::detail {
+template<int ElementSize, int BlockWidth, int BlockHeight, int BlockCount>
+struct XeSubgroup2DBlockLoad {
+  template<typename T>
+  CUTE_HOST_DEVICE
+  void operator()(const void *srcBasePointer, int memoryWidth, int memoryHeight, int memoryPitch,
+                  cute::intel::coord_t coordinate, T *dstPointer) {
+    __spirv_Subgroup2DBlockLoadINTEL(ElementSize, BlockWidth, BlockHeight, BlockCount,
+                                     srcBasePointer, memoryWidth,
+                                     memoryHeight, memoryPitch, coordinate,
+                                     static_cast<void *>(dstPointer));
+  }
+};
+
+template<int ElementSize, int BlockWidth, int BlockHeight, int BlockCount>
+struct XeSubgroup2DBlockTransform {
```

Comment:
In my opinion, `XeSubgroup2DBlockTransform` and `XeSubgroup2DBlockTranspose` would be better named `XeSubgroup2DBlockLoadTransform`  and `XeSubgroup2DBlockLoadTranspose`. Otherwise, it sounds like the function simply transforms/transposes a block.